requirements.txt

```diff
@@ -1,23 +1,18 @@
-connexion[swagger-ui]==2.7.0
-email-validator
-fasteners
-flask==1.1.2
-flask-authorize
-flask-jwt-extended>=4.1.0
-flask_bootstrap==3.3.7.1
-flask_cors==3.0.8
-flask_debugtoolbar==0.11.0
-flask_login==0.5.0
-flask_mail==0.9.1
-flask_marshmallow==0.12.0
-flask_moment==0.9.0
-flask_sqlalchemy==2.4.1
-flask_wtf==0.14.3
-marshmallow==3.5.1
-marshmallow_sqlalchemy==0.23.0
-python_dateutil==2.8.1
-requests=2.23.0
-seamm_util
-selenium==3.141.0
-sqlalchemy==1.3.16
-wtforms==2.3.1
+connexion==2.7.0
+fasteners==0.16.3
+Flask==2.0.2
+Flask_Authorize==0.2.6
+Flask_Bootstrap==3.3.7.1
+Flask_Cors==3.0.10
```

Comment:
I'm not sure if we actually use this.

Reply:
I agree and have been thinking about this. In general two make sense: what is now called `requirements_install.txt`, which contains the unpinned requirements, and `requirements_dev.txt` which adds the requirements such as `black`, `flake8` and `purest` needed for development. `requirements_dev.txt` includes `requirements_install.txt` so it truly does add just what is needed for development but is not needed to run.

`requirements.txt` contains pinned requirements, nominally the actual version used in that build and test. It cam from my early reading (I didn't use Python before joining MolSSI) that this was how you should handle requirements, with lots of instructions to use `pip list` to generate it. Obviously using `pip list` is way overkill and not useful, and I am with you thinking that this is just not useful -- and it is a pain to maintain! The information is available in the Github Actions logs if it is even needed, though I suspect it never will be.

So...we should remove it and it would be best to rename `requirements_install.txt` to plain `requirements.txt`. This will involve a small edit in `setup.py` and in `requirements.dev` (unless we want to merge the two, which is not, I think, a good idea). Unfortunately SEAMM now has almost 40 repositories, and it is preferable to keep them as similar as possible, so when I have a few hours -- perhaps after too much turkey when the brain is not working too well -- I'll change them all, make 40 PRs and then merge them all. A bit tedious, but such is life. 

What do you think?

Reply:
As an aside, for most of our modules installing from PyPi is quite sensible and I have not yet put most on CondaForge, and am not sure that it is worth the effort to do so. The Dashboard is different (and maybe the Datastore?) in that it can't really be installed using `pip`. At least I think not, but perhaps now that you filter the node.js files and we include them in the release it actually does work. 

If it doesn't make sense, then neither does the requirements file, and perhaps we should see if we can remove the Dashboard from PyPi and build the conda version directly from GitHub releases, which I think can easily be done. It would be different than the other modules, where CondaForge automatically builds from PyPi releases, but that would not be too troublesome. We'd just have to remember that the Dashboard is different and handle it properly. 

Thoughts?

I am afraid that this is the reality of managing a large, sprawling project like SEAMM. It is usually a bit of a compromise and not as elegant as you'd like, tends to be a lot of tedious work, and consistency is very important (hence the compromise).

Reply:
I think the datastore should be fully installable using pip. The dashboard as well if we filter include the needed npm files. I believe this is what will need to be done for conda as well. I don't know if I quite understand anything about this, but my preference would be to keep it as similar to the other projects as possible. The "build" portion for the node files is really just a way for us to easily update the packages if needed.

Reply:
I also am not really up on the best practices with requirement files. My feeling is that we shouldn't need 3, but I'd have to look into Python documentation to see what's recommended.